### PR TITLE
WT-7657 Update comment about OOO/mixed mode evictions and the history store.

### PR DESCRIPTION
**Pull request by sauclovian-wt**
The other concerns in WT-7657 are covered by other tickets. See the discussion in the ticket.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

**Review by quchenhao — APPROVED** [review]
lgtm

File: src/reconcile/rec_write.c [hunk 1]
```diff
@@ -2738,10 +2738,12 @@ __wt_rec_hs_clear_on_tombstone(WT_SESSION_IMPL *session, WT_RECONCILE *r, wt_tim
         WT_RET(__wt_curhs_open(session, NULL, &r->hs_cursor));
 
     /*
-     * From WT_TS_NONE delete all the history store content of the key. This path will never be
-     * taken for a mixed-mode deletion being evicted and with a checkpoint that started prior to the
-     * eviction starting its reconciliation as previous checks done while selecting an update will
-     * detect that.
+     * From WT_TS_NONE delete all the history store content of the key. The test of
```
Comment (kommiharibabu) [inline src/reconcile/rec_write.c:2741]: Currently, the `__wt_rec_hs_clear_on_tombstone` is called for reinsert scenarios also when the 0 timestamp delete is not yet globally visible. 

How about updating the statement as `From WT_TS_NONE delete/reinsert all ...`?

**Review by kommiharibabu — APPROVED** [review]
LGTM. A small comment update suggestion.

File: src/reconcile/rec_write.c (hunk 1 above)
Comment (sauclovian-wt) [reply src/reconcile/rec_write.c]: Sure. Fixed in [21f2f3d](https://github.com/wiredtiger/wiredtiger/pull/7743/commits/21f2f3d4704b6b87833c3307b33887625482deba).

**Pull request merged**